spring-ai-core/src/main/java/org/springframework/ai/chat/model/AbstractToolCallSupport.java

```diff
@@ -141,12 +142,22 @@ protected List<Message> handleToolCalls(Prompt prompt, ChatResponse response) {
 		Map<String, Object> toolContextMap = Map.of();
 		if (prompt.getOptions() instanceof FunctionCallingOptions functionCallOptions
 				&& !CollectionUtils.isEmpty(functionCallOptions.getToolContext())) {
```

Comment:
This logic would prevent the conversion history to be inserted if the tool context was empty from chat client request spec.